src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -32,23 +51,82 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
         // monarch!
         CoRevokeClassObject(_registrationHostClass);
         _registrationHostClass = 0;
+        _monarchWaitInterrupt.SetEvent();
+
+        // A thread is joinable once it's been started. Basically this just
+        // makes sure that the thread isn't just default-constructed.
+        if (_electionThread.joinable())
+        {
+            _electionThread.join();
+        }
     }
 
     void WindowManager::ProposeCommandline(const Remoting::CommandlineArgs& args)
     {
-        const bool isKing = _areWeTheKing();
         // If we're the king, we _definitely_ want to process the arguments, we were
         // launched with them!
         //
         // Otherwise, the King will tell us if we should make a new window
-        _shouldCreateWindow = isKing ||
-                              _monarch.ProposeCommandline(args);
+        _shouldCreateWindow = _isKing;
+        std::optional<uint64_t> givenID;
+        if (!_isKing)
+        {
+            // The monarch may respond back "you should be a new
+            // window, with ID,name of (id, name)". Really the responses are:
+            // * You should not create a new window
+            // * Create a new window (but without a given ID or name). The
+            //   Monarch will assign your ID/name later
+            // * Create a new window, and you'll have this ID or name
+            //   - This is the case where the user provides `wt -w 1`, and
+            //     there's no existing window 1
+
+            const auto result = _monarch.ProposeCommandline(args);
+            _shouldCreateWindow = result.ShouldCreateWindow();
+            if (result.Id())
+            {
+                givenID = result.Id().Value();
+            }
+
+            // TraceLogging doesn't have a good solution for logging an

```

Comment:
Not only that, the trace logging macros explicitly tell you "DO NOT BE CREATIVE". So this is fine.